pkg/kubelet/util/manager/cache_based_manager.go

```diff
@@ -175,7 +174,8 @@ func (s *objectStore) Get(namespace, name string) (runtime.Object, error) {
 	// needed and return data.
 	data.Lock()
 	defer data.Unlock()
-	if data.err != nil || !s.isObjectFresh(data) {
+	refUpdated := data.lastVersion < refVersion
+	if data.err != nil || !s.isObjectFresh(data, refUpdated) {
```

Comment:
nit: Negate this conditional & shortcircuit, to make the control flow easier to follow.